node/src/accountant/db_big_integer/big_int_db_processor.rs

```diff
@@ -408,61 +442,59 @@ impl StdNumParamFormNamed {
     }
 }
 
-pub struct Param<'a> {
-    value_pair: (&'a str, &'a dyn ExtendedParamsMarker),
-    participates_in_overflow_clause: bool,
+pub enum Param<'params> {
+    BothClauses(DisplayableRusqliteParamPair<'params>),
+    MainClauseLimited(DisplayableRusqliteParamPair<'params>),
 }
 
-impl<'a> Param<'a> {
-    pub fn new(
-        value_pair: (&'a str, &'a (dyn ExtendedParamsMarker + 'a)),
-        participates_in_overflow_clause: bool,
-    ) -> Self {
-        Self {
-            value_pair,
-            participates_in_overflow_clause,
+impl<'params> Param<'params> {
+    fn param_pair(&self) -> &DisplayableRusqliteParamPair<'params> {
+        match self {
+            Param::BothClauses(pair) => pair,
+            Param::MainClauseLimited(pair) => pair,
         }
     }
 
-    fn as_rusqlite_params(&'a self) -> (&'a str, &'a dyn ToSql) {
-        (self.value_pair.0, &self.value_pair.1 as &dyn ToSql)
+    fn as_rusqlite_param_pair(&'params self) -> RusqliteParamPair<'params> {
+        let pair = self.param_pair();
+        (pair.0, &pair.1 as &dyn ToSql)
```

Comment:
If it was a struct instead of a tuple, it would be easier to understand what the first and the second element of the tuple mean. Also, a reference can be removed if we use a Box.